validphys2/src/validphys/lhapdfset.py

```diff
@@ -0,0 +1,138 @@
+    4: 0.002180124185625795,
+    5: 6.922722705177504e-05,
+    21: 0.007604124516892057}
+"""
+import logging
+import numpy as np
+import lhapdf
+
+log = logging.getLogger(__name__)
+
+
+class LHAPDFSet:
+    """Wrapper for the lhapdf python interface.
+
+    Once instantiated this class will load the PDF set from LHAPDF.
+    If it is a T0 set only the CV will be loaded.
+
+    For Monte Carlo sets the central value (member 0) is by default not included when taking
+    the results for all members (i.e., when using ``grid_values``).
+    """
+
+    def __init__(self, name, error_type):
+        log.info("PDF: %s ErrorType: %s", name, error_type)
+        self._name = name
+        self._error_type = error_type
+        if self.is_t0:
+            # If at this point we already know this is a T0 set, load only the CV
+            self._lhapdf_set = [lhapdf.mkPDF(name)]
+        else:
+            self._lhapdf_set = lhapdf.mkPDFs(name)
+        self._flavors = None
+
+    @property
+    def is_monte_carlo(self):
+        """Check whether the error type is MC"""
+        return self._error_type == "replicas"
+
+    @property
+    def is_t0(self):
+        """Check whether we are in t0 mode"""
+        return self._error_type == "t0"
+
+    @property
+    def n_members(self):
+        """Return the number of active members in the PDF set"""
+        return len(self.members)
+
+    @property
+    def members(self):
+        """Return the members of the set, this depends on the error type:
+        t0: returns only member 0
+        MC: skip member 0
+        Hessian: return all
+        """
+        if self.is_t0:
+            return self._lhapdf_set[0:1]
+        if self.is_monte_carlo:
+            return self._lhapdf_set[1:]
+        return self._lhapdf_set
+
+    @property
+    def central_member(self):
+        """Returns a reference to member 0 of the PDF list"""
+        return self._lhapdf_set[0]
+
+    def xfxQ(self, x, Q, n, fl):
+        """Return the PDF value for one single point for one single member
+        If the flavour is not included in the PDF (for instance top/antitop) return 0.0
+        """
+        member_pdf = self.members[n]
+        res = member_pdf.xfxQ(x, Q)
+        return res.get(fl, 0.0)
+
+    @property
+    def flavors(self):
+        """Returns the list of accepted flavors by the LHAPDF set"""
+        if self._flavors is None:
+            self._flavors = self.members[0].flavors()
+        return self._flavors
+
+    def grid_values(self, flavors: np.ndarray, xgrid: np.ndarray, qgrid: np.ndarray):
+        """Returns the PDF values for every member for the required
+        flavours, points in x and pointx in q
+        The return shape is
+            (members, flavors, xgrid, qgrid)
+        Return
+        ------
+            ndarray of shape (members, flavors, xgrid, qgrid)
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from validphys.lhapdfset import LHAPDFSet, ER_MC
```

Comment:
ImportError: cannot import name 'ER_MC' from 'validphys.lhapdfset' (/home/zah/nngit/nnpdf/validphys2/src/validphys/lhapdfset.py)

Reply:
And rightly so. Sorry, I copied the example from the closed PR and missed that. I'll fix it later today.